### ISSUE TITLE
Example phones/ OS versions that support enabling IMS

### ISSUE THREAD

**Issue by nishant412** [herlesupreeth/docker_open5gs#372]
I have been successful setting up the docker images for Open5Gs core + Kaimilio IMS, and have been able to register a OnePlus 7 Pro running Lineage OS Android 11, using the CoIMS app method. Unfortunately I havent had much luck in finding an alternate phone/OS version that works for me. I have tried a Samsung Galacy S9 (Android 10 - Samsung setting menu in CoIMS doesnt launch), Xiaomi Redmi 13C (Android 13 - unable to launch CoIMS app), Iphone 12 (iOS 17.4 - no real way to set an IMS apn), Pixel 7 (Android 13 - samsung chipset doesnt let launch the samsung CoIMS settings). 

Are there any suggestions for phones/OS that can work? Will just rooting a phone with custom ROM lineage OS help or the settings that are locked down are from the baseband firmware side? 

Really appreciate any suggestions

Thanks in advance

**Comment by herlesupreeth** [comment]
I wouldnt recommend Samsung phones as I havent found a way to enable IMS on them after Android 10 update.

> Xiaomi Redmi 13C (Android 13 - unable to launch CoIMS app

I am not sure what' the issue here. As long as the SIM has the SHA1 key for the app the CoIMS app should load.

> Iphone 12 (iOS 17.4 - no real way to set an IMS apn)

No need to add ims APN for iPhones. IMS is enabled by default. Atleast I have tried with iPhone 11, iPhone 13 pro with PLMN 001 01

> Pixel 7 (Android 13 - samsung chipset doesnt let launch the samsung CoIMS settings)

For Pixel phones, its enough that you run CoIMS app once and IMS should be enabled. You dont need to go to Samsung IMS settings submenu

**Comment by nishant412** [comment]
Thanks for your response. I have been trying to connect using the iphone 12 but unable to get IMS connectivity. The internet APN seems to connect just fine. Here are the pcap logs of the iphone turning on and connecting:
[iphone_noims.zip](https://github.com/user-attachments/files/17062244/iphone_noims.zip)

Would really appreciate any inputs on what I am doing wrong.
Thanks in advance.

**Comment by herlesupreeth** [comment]
You are right. iPhone is not initiating establishment of "ims" PDN. Can you please check whether you have enabled under Settings -> Cellular -> SIM -> Voice & Data --> VoLTE

**Comment by nishant412** [comment]
Sorry for the late response. Thank you for the comment. The iPhone did indeed have VoLTE turned off, so I enabled that and it is connecting to IMS now. Also was able to get the Redmi phone to connect as well, even though the CoIMS seems to be crashing on launch. I havent tested the Pixel again so far, but I am hopeful that it should work too.

Lastly I observed that updating the WebUI with subscriber info and pyHSS does not seem to take effect live, and I need to restart all my docker containers for the phones to actually connect to IMS (they connect to data right away without needing to restart). Dunno if this is by design, but its no big deal for now.

Thanks for your help with this.
